### PR TITLE
[EventHubs] Fix Token Decode Error

### LINKED ISSUE #25016: [Python AMQP] update token_type being passed into JWTTokenAuth
This like looks wrong: https://github.com/Azure/azure-sdk-for-python/blob/1ad3a2c8d7149838d5f3b45c3085cd5b4161583a/sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/authentication.py#L128

Should be popping `token_type` I believe. However, replacing "kwargs" with "token_type" while using the DefaultAzureCredential results in the following error:
```
Traceback (most recent call last):
  File "c:\users\swathip\documents\forks\azure-sdk-for-python\sdk\eventhub\azure-eventhub\azure\eventhub\_client_base.py", line 572, in _do_retryable_operation
    return operation()
  File "c:\users\swathip\documents\forks\azure-sdk-for-python\sdk\eventhub\azure-eventhub\azure\eventhub\_client_base.py", line 529, in _open
    while not self._handler.client_ready():
  File "c:\users\swathip\documents\forks\azure-sdk-for-python\sdk\eventhub\azure-eventhub\azure\eventhub\_pyamqp\client.py", line 317, in client_ready
    if not self.auth_complete():
  File "c:\users\swathip\documents\forks\azure-sdk-for-python\sdk\eventhub\azure-eventhub\azure\eventhub\_pyamqp\client.py", line 304, in auth_complete
    if self._cbs_authenticator and not self._cbs_authenticator.handle_token():        
  File "c:\users\swathip\documents\forks\azure-sdk-for-python\sdk\eventhub\azure-eventhub\azure\eventhub\_pyamqp\cbs.py", line 221, in handle_token
    raise TokenAuthFailure(
azure.eventhub._pyamqp.error.TokenAuthFailure: CBS Token authentication failed.
Status code: 400
Description: Property 'type' in the 'application-properties' section is either missing, has the wrong type, or has an unexpected value.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\swathip\Documents\forks\azure-sdk-for-python\sdk\eventhub\azure-eventhub\samples\sync_samples\client_identity_authentication.py", line 61, in <module>     
    event_data_batch = producer.create_batch()
  File "c:\users\swathip\documents\forks\azure-sdk-for-python\sdk\eventhub\azure-eventhub\azure\eventhub\_producer_client.py", line 350, in create_batch
    self._get_max_mesage_size()
  File "c:\users\swathip\documents\forks\azure-sdk-for-python\sdk\eventhub\azure-eventhub\azure\eventhub\_producer_client.py", line 129, in _get_max_mesage_size
    cast(
  File "c:\users\swathip\documents\forks\azure-sdk-for-python\sdk\eventhub\azure-eventhub\azure\eventhub\_producer.py", line 154, in _open_with_retry
    return self._do_retryable_operation(self._open, operation_need_param=False)       
  File "c:\users\swathip\documents\forks\azure-sdk-for-python\sdk\eventhub\azure-eventhub\azure\eventhub\_client_base.py", line 588, in _do_retryable_operation
    raise last_exception
azure.eventhub.exceptions.AuthenticationError: CBS Token authentication failed.       
Status code: 400
Description: Property 'type' in the 'application-properties' section is either missing, has the wrong type, or has an unexpected value.
Error: client-error
CBS Token authentication failed.
Status code: 400
Description: Property 'type' in the 'application-properties' section is either missing, has the wrong type, or has an unexpected value.
```

### LINKED ISSUE #27024: [Event Hub] Deploy test resources failed: DatabaseAccount is in a failed provisioning state.
We are running live Tests against other clouds like US Gov and Azure China Cloud. The goal is to check whether new azure sdk package work with other clouds or not.

**Error Description:**
Deploying test resources error occurred in all environments under `China cloud`. The error message was shown as below: 
```
Retry : Exception calling "Invoke" with "0" argument(s): "The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: 7:54:28 AM - The deployment 't78cea9bbb7ee4bb3' failed with error(s). Showing 1 out of 1 error(s).
Status Message: DatabaseAccount storaget78cea9bbb7ee4bb3 is in a failed provisioning state because the previous attempt to create it was not successful. Please delete the previous instance before attempting to recreate this account.
ActivityId: 11ae189e-d9a0-4e78-85db-4880e6425870, Microsoft.Azure.Documents.Common/2.14.0 (Code:BadRequest)
CorrelationId: c1e7ab97-b123-4bf3-ab80-2dbab5210d28"
At D:\a\_work\1\s\eng\common\TestResources\New-TestResources.ps1:725 char:23
+         $deployment = Retry {
+                       ~~~~~~~
+ CategoryInfo          : NotSpecified: (:) [Write-Error], MethodInvocationException
+ FullyQualifiedErrorId : ActionPreferenceStopException,Retry
##[error]PowerShell exited with code '1'.
```

For more details, please click [here](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1938773&view=logs&j=6e82a61b-0a7c-58a1-7780-0f3eac3f7007&t=a492cae6-8134-576d-93bd-8adc52b09d52&l=13620).

**Expected Behavior:**
The resource can be deployed successfully on `China` cloud.

@benbp , @scottaddie , @jsquire , @ronniegeraghty , @annatisch for notification.

### LINKED ISSUE #28264: [EventHubs] enable canary regions in the test pipeline

### PR DESCRIPTION
**Pull request by kashifkhan**
Fixes #28264

We ran in to an issue where we were seeing 401 unauthorized errors on get_partition_ids, get_eventhub_properties etc. on Canary

Canary has turned off anonymous access for those get partition calls. In general, public EH has this now enabled globally except for North America. Canary has enabled this everywhere.

The service is expecting a string token, but our code was sending an encoded token only if `generate_sas_token` was called ie when using connection string based calls. This affected management calls, as the service saw those tokens as empty and therefore treated them as anonymous calls.

Additionally, updated the `AccessToken.token` stored in `EventHubSharedKeyCredential` (returned by internal `_generate_sas_token`) to return a string and not bytes. This was a bug introduced in 5.10.1, when the uamqp switch was added.

Removed Cosmos from `test-resources.json`, which resulted in a deployment error due to limits on cosmos resource in canary regions. This should also fix #27024, fix #25016

### CODE REVIEW

File: .vscode/cspell.json [hunk 1]
```diff
@@ -165,6 +165,7 @@
     "dtlksd",
     "DWORD",
     "eastus",
+    "eaup",
```
Comment (swathipil) [inline .vscode/cspell.json:168]: euap 🤦

**Review by annatisch — COMMENTED** [review]
Changes look okay to me.
Is the change in return type of `credential.get_token()` public?
Wondering if it's avoidable?

**Review by annatisch — APPROVED** [review]

**Pull request merged**